packages/cqrs/lib/src/cqrs_result.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 LeanCode Sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:equatable/equatable.dart';
+
+import 'command_result.dart';
+import 'cqrs_error.dart';
+
+/// Generic result for CQRS query result. Can be either [QSuccess]
+/// or [QFailure].
+sealed class QResult<T> extends Equatable {
```

Comment:
Can we use full name instead of a single letter in the results? I don't like it and I think `OperationFailure` isn't _that_ much longer that `OFailure`.

When we have an if-case 3 levels deep (e.g. class -> method -> try) we have 32 characters on the line left for the variable name we are casing on in case of longest `OperationFailure`:

```dart
class A {
  void a() {
    try {
      if (a case CommandFailure(:final data)) {/*                             */
    }
  }
```

Reply:
I agree and I'm going to change that. However, this would cause a minor conflict with already existing `CommandResult`. My solution to that is to rename the already existing `CommandResult` to `CommandResponse`. What do you think?